src/reducers/url-state.js

```diff
@@ -83,25 +83,12 @@ function selectedThread(
 ): ThreadIndex | null {
   switch (action.type) {
     case 'CHANGE_SELECTED_THREAD':
-      return action.selectedThread;
     case 'VIEW_PROFILE':
+    case 'ISOLATE_GLOBAL_TRACK':
+    case 'HIDE_GLOBAL_TRACK':
+    case 'HIDE_LOCAL_TRACK':
+    case 'ISOLATE_LOCAL_TRACK':
```

Comment:
I'm a bit concerned that so much logic is in the action and not the reducer. I feel like we don't use redux correctly and this may hit us later. (although I can't really say how now).